host/greatfet/commands/greatfet_chipcon.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+#
+# This file is part of GreatFET
+
+from __future__ import print_function
+
+import errno
+import sys
+import ast
+
+import greatfet
+from greatfet import GreatFET
+from greatfet.utils import log_silent, log_verbose
+from greatfet.programmers.chipcon import ChipconProgrammer
+
+
+def main():
+    from greatfet.utils import GreatFETArgumentParser
+
+    # Set up a simple argument parser.
+    parser = GreatFETArgumentParser(description="Utility for chipcon debugging via GreatFET")
+    parser.add_argument('-id', '--chip_id', action='store_true',
+                        help="Print the chip ID of the connected device.")
+    parser.add_argument('-a', '--address', nargs=1, type=ast.literal_eval,
+                        help="Flash memory address to begin reading from / writing to")
+    parser.add_argument('-r', '--read', type=int,
+                        help="Number of bytes to read from flash memory")
+    parser.add_argument('-w', '--write_page', nargs='*', type=ast.literal_eval,
+                        help="Data to write to the specified flash page")
+    parser.add_argument('-p', '--program_flash',
+                        help="File name from which data will be written to flash from")
+    parser.add_argument('-e', '--erase', default=True,
```

Comment:
If the default is already `True`, than specifying `--erase` does nothing. But erasing by default makes sense, so an option to negate it is needed. Add `action=store_false`, and change the option to `--no-erase`.